golang/cosmos/x/vstorage/vstorage.go

```diff
@@ -57,46 +66,89 @@ func (sh vstorageHandler) Receive(cctx *vm.ControllerContext, str string) (ret s
 	// Handle generic paths.
 	switch msg.Method {
 	case "set":
-		keeper.SetStorageAndNotify(cctx.Context, msg.Path, msg.Value)
+		for _, arg := range msg.Args {
+			var entry types.StorageEntry
+			entry, err = types.UnmarshalStorageEntry(arg)
+			if err != nil {
+				return
+			}
+			keeper.SetStorageAndNotify(cctx.Context, entry)
+		}
 		return "true", nil
 
 		// We sometimes need to use LegacySetStorageAndNotify, because the solo's
 		// chain-cosmos-sdk.js consumes legacy events for `mailbox.*` and `egress.*`.
 		// FIXME: Use just "set" and remove this case.
 	case "legacySet":
-		//fmt.Printf("giving Keeper.SetStorage(%s) %s\n", msg.Key, msg.Value)
-		keeper.LegacySetStorageAndNotify(cctx.Context, msg.Path, msg.Value)
+		for _, arg := range msg.Args {
+			var entry types.StorageEntry
+			entry, err = types.UnmarshalStorageEntry(arg)
+			if err != nil {
+				return
+			}
+			//fmt.Printf("giving Keeper.SetStorage(%s) %s\n", entry.Path(), entry.Value())
+			keeper.LegacySetStorageAndNotify(cctx.Context, entry)
+		}
 		return "true", nil
 
 	case "setWithoutNotify":
-		keeper.SetStorage(cctx.Context, msg.Path, msg.Value)
+		for _, arg := range msg.Args {
+			var entry types.StorageEntry
+			entry, err = types.UnmarshalStorageEntry(arg)
+			if err != nil {
+				return
+			}
+			keeper.SetStorage(cctx.Context, entry)
+		}
 		return "true", nil
 
 	case "append":
-		err = keeper.AppendStorageValueAndNotify(cctx.Context, msg.Path, msg.Value)
-		if err != nil {
-			return "", err
+		for _, arg := range msg.Args {
+			var entry types.StorageEntry
+			entry, err = types.UnmarshalStorageEntry(arg)
+			if err != nil {
+				return
+			}
+			if !entry.HasData() {
+				err = errors.New("No value for append entry with path: " + entry.Path())
+				return
+			}
+			err = keeper.AppendStorageValueAndNotify(cctx.Context, entry.Path(), entry.StringValue())
+			if err != nil {
+				return "", err
```

Comment:
Style consistency nit:
```suggestion
				return
```